extensions/sql-migration/src/wizard/accountsSelectionPage.ts

```diff
@@ -0,0 +1,112 @@
+		this._accountsMap = new Map();
+	}
+
+	protected async registerContent(view: azdata.ModelView): Promise<void> {
+		const form = view.modelBuilder.formContainer().withFormItems(
+			[
+				await this.createAzureAccountsDropdown(view)
+			]
+		);
+		await view.initializeModel(form.component());
+		await this.populateAzureAccountsDropdown();
+		return;
+	}
+
+	private async createAzureAccountsDropdown(view: azdata.ModelView): Promise<azdata.FormComponent> {
+
+		this._azureAccountsDropdown = view.modelBuilder.dropDown().withProperties({}).component();
+
+		this._azureAccountsDropdown.onValueChanged(async (value) => {
+			this.migrationStateModel.azureAccount = this._accountsMap.get(value.selected)!;
+		});
+
+		const addAccountButton = view.modelBuilder.button().withProperties<azdata.ButtonProperties>({
+			label: constants.ACCOUNT_ADD_BUTTON_LABEL,
+			width: '100px'
+		}).component();
+
+		addAccountButton.onDidClick(async (event) => {
+			await vscode.commands.executeCommand('workbench.actions.modal.linkedAccount');
+			await this.populateAzureAccountsDropdown();
+		});
+
+		const flexContainer = view.modelBuilder.flexContainer().withLayout({
+			flexFlow: 'column'
+		}).withItems([this._azureAccountsDropdown, addAccountButton], { CSSStyles: { 'margin': '10px', } }).component();
+
+		return {
+			title: '',
+			component: flexContainer
+		};
+	}
+
+	private async populateAzureAccountsDropdown() {
+		this._azureAccountsDropdown.loading = true;
+		let accounts = await azdata.accounts.getAllAccounts();
+
+		if (accounts.length === 0) {
+			this._azureAccountsDropdown.value = constants.ACCOUNT_SELECTION_PAGE_NO_LINKED_ACCOUNTS_ERROR;
+			return;
+		}
+
+		this._azureAccountsDropdown.values = accounts.map((account): azdata.CategoryValue => {
+			let accountCategoryValue = {
+				displayName: account.displayInfo.displayName,
+				name: account.displayInfo.displayName
+			};
+			this._accountsMap.set(accountCategoryValue.displayName, account);
+			return accountCategoryValue;
+		});
+
+		this.migrationStateModel.azureAccount = accounts[0];
+		this._azureAccountsDropdown.loading = false;
+	}
+
+	public onPageEnter(): Promise<void> {
+		this.wizard.registerNavigationValidator((pageChangeInfo: azdata.window.WizardPageChangeInfo) => {
+			if (pageChangeInfo.newPage < pageChangeInfo.lastPage) {
+				return true;
+			}
+
+			if (this._azureAccountsDropdown.value === constants.ACCOUNT_SELECTION_PAGE_NO_LINKED_ACCOUNTS_ERROR) {
+				this.wizard.message = {
+					text: constants.ACCOUNT_SELECTION_PAGE_NO_LINKED_ACCOUNTS_ERROR,
+					level: azdata.window.MessageLevel.Error
+				};
+				return false;
+			}
+			return true;
+		});
+
+		return Promise.resolve();
+	}
+
+	public onPageLeave(): Promise<void> {
+		this.wizard.registerNavigationValidator((pageChangeInfo: azdata.window.WizardPageChangeInfo) => {
+			return true;
+		});
+		return Promise.resolve();
+	}
+
+	protected handleStateChange(e: StateChangeEvent): Promise<void> {
```

Comment:
Why do we even need to implement this method in this subclass if it is always going to throw?

Reply:
I will implement this method when other pages of the wizard are done.

Reply:
Why would this throw though? I would expect it to no-op instead.